dialogue-clients/src/main/java/com/palantir/dialogue/clients/ChannelNames.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.clients;
+
+final class ChannelNames {
+
+    static String reloading(String serviceName, ImmutableReloadingParams params) {
+        return "dialogue-" + serviceName + summarizeOptions(params);
+    }
+
+    static String nonReloading(Class<?> clazz, AugmentClientConfig augment) {
+        return "dialogue-nonreloading-" + clazz.getSimpleName() + summarizeOptions(augment);
+    }
+
+    private static String summarizeOptions(AugmentClientConfig augment) {
+        StringBuilder builder = new StringBuilder();
+        augment.nodeSelectionStrategy().ifPresent(value -> builder.append("-").append(value));
+        augment.maxNumRetries()
+                .ifPresent(value -> builder.append("-MAX_RETRIES_").append(value));
+        augment.clientQoS().ifPresent(value -> builder.append("-").append(value));
+        augment.serverQoS().ifPresent(value -> builder.append("-").append(value));
+        augment.retryOnTimeout().ifPresent(value -> builder.append("-").append(value));
```

Comment:
I think this has caused some confusion, thoughts on updating slightly?
```suggestion
        augment.retryOnTimeout().ifPresent(value -> builder.append("-").append(value ? "retry-timeouts" : "no-timeout-retries"));
```

Reply:
It's actually already a nice enum, so should show up in the toString properly. I think the `-true` / `-false` suffix in the atlasdb PR was something else

```
enum RetryOnTimeout {
        /** Default. */
        DISABLED,
        /**
         * Enables retry on read/write timeout. This was the default behavior in versions up to and including 4.24.x.
         * Consumers should almost never use this option, reserving it for when their clients have guaranteed that they
         * will never retry on timeout.
         *
         * <p>The risk of retry storms is severe. If the client times out before the consumer finishes retrying on its
         * timeout, it will submit a new request which will also retry on timeout. Expensive requests that time out have
         * brought down servers with this behavior enabled.
         *
         * <p>Note that connect timeouts will always be retried.
         */
        DANGEROUS_ENABLE_AT_RISK_OF_RETRY_STORMS
    }
```

Reply:
Ah you’re right, that was the atlas short-timeout flag